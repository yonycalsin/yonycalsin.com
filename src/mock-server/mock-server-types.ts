import type { DefaultBodyType, PathParams, ResponseComposition, RestContext, RestRequest } from 'msw'

export type MockRequest<Body = never, Params extends PathParams = PathParams<string>> = RestRequest<Body, Params>

export type MockResponse = ResponseComposition<DefaultBodyType>

export type MockContext = RestContext
