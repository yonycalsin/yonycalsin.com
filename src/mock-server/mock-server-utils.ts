function sanitizeEndpoint(endpoint: string) {
  const url = new URL(endpoint)

  if (!url.search) {
    return url.href
  }

  return url.href.replace(url.search, '')
}

export { sanitizeEndpoint }
