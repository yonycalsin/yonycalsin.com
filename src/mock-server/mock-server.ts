export async function initMocks() {
  if (typeof window === 'undefined') {
    const { server } = await import('./mock-server-node')

    server.listen()
  } else {
    const { worker } = await import('./mock-server-browser')

    worker.start()
  }
}
