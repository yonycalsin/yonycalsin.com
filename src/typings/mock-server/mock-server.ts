import type { DefaultBodyType, PathParams, ResponseComposition, RestContext, RestRequest } from 'msw'

type MockRequest<Body extends DefaultBodyType = never, Params extends PathParams = PathParams<string>> = RestRequest<
  Body,
  Params
>

type MockResponse = ResponseComposition<DefaultBodyType>

type MockContext = RestContext

export type { MockContext, MockRequest, MockResponse }
