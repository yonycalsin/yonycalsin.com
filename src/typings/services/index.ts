export * from './blog'
export * from './shared'
