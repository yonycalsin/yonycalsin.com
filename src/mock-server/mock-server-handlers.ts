import { rest } from 'msw'

import { API_ENDPOINTS } from 'services/shared'
import {
  getAllAchievementsApi,
  getAllBooksApi,
  getAllPagesApi,
  getAllProjectsApi,
  getCategoriesApi,
  getCategoryApi,
  getFeaturedAchievementsApi,
  getFeaturedRecommendationsApi,
  getPageApi,
  getPinnedProjectsApi,
  getPostApi,
  getPostsApi,
  getReadingBooksApi,
} from './mocks'

export const mockServerHandlers = [
  // Achievements
  rest.get(API_ENDPOINTS.ALL_ACHIEVEMENTS, getAllAchievementsApi),
  rest.get(API_ENDPOINTS.FEATURED_ACHIEVEMENTS, getFeaturedAchievementsApi),

  // Posts
  rest.get(API_ENDPOINTS.POST(':slug'), getPostApi),
  rest.get(API_ENDPOINTS.POSTS, getPostsApi),

  // Categories
  rest.get(API_ENDPOINTS.CATEGORY(':slug'), getCategoryApi),
  rest.get(API_ENDPOINTS.CATEGORIES, getCategoriesApi),

  // Books
  rest.get(API_ENDPOINTS.ALL_BOOKS, getAllBooksApi),
  rest.get(API_ENDPOINTS.READING_BOOKS, getReadingBooksApi),

  // Pages
  rest.get(API_ENDPOINTS.PAGE(':slug'), getPageApi),
  rest.get(API_ENDPOINTS.ALL_PAGES, getAllPagesApi),

  // Projects
  rest.get(API_ENDPOINTS.ALL_PROJECTS, getAllProjectsApi),
  rest.get(API_ENDPOINTS.PINNED_PROJECTS, getPinnedProjectsApi),

  // Recommendations
  rest.get(API_ENDPOINTS.FEATURED_RECOMMENDATIONS, getFeaturedRecommendationsApi),
]
