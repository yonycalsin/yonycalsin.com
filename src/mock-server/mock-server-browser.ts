import { setupWorker } from 'msw'

import { mockServerHandlers } from './mock-server-handlers'

export const worker = setupWorker(...mockServerHandlers)
