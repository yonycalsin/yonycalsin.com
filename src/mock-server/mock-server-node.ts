import { setupServer } from 'msw/node'

import { mockServerHandlers } from './mock-server-handlers'

function initServer() {
  const server = setupServer(...mockServerHandlers)

  return server
}

export default initServer
