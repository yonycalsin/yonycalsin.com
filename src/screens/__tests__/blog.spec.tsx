import { screen, within } from '@testing-library/react'

import { postsSuccess } from 'mock-server/mocks'
import { setupWithReactQuery, TestProvider } from 'tests'
import { getPostsApi } from 'services'
import BlogScreen from 'screens/blog'

jest.mock('services/posts')

const mockGetPostsApi = getPostsApi as jest.Mock

function setup() {
  return setupWithReactQuery(
    <TestProvider>
      <BlogScreen />
    </TestProvider>,
  )
}

describe('BlogScreen', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('renders the blog screen', async () => {
    mockGetPostsApi.mockReturnValueOnce(postsSuccess)

    const view = setup()

    await screen.findByRole('progressbar')

    const list = await screen.findByRole('list', {
      name: /list of posts/i,
    })

    const items = within(list).getAllByRole('listitem')

    expect(items).toHaveLength(items.length)

    const title = screen.queryByRole('heading', { name: `Posts (${postsSuccess.data.length})` })

    expect(title).toBeInTheDocument()

    expect(view.baseElement).toMatchSnapshot()
  })
})
