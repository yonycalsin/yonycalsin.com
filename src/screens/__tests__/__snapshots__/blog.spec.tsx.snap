// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogScreen renders the blog screen 1`] = `
<body>
  <div>
    <div
      class="w-full z-10 shadow-sm"
    >
      <div
        class="container py-5 md:py-6 flex justify-between items-center"
      >
        <a
          href="/"
        >
          <img
            alt="Logo"
            data-nimg="1"
            decoding="async"
            height="32"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
            style="color: transparent;"
            width="32"
          />
        </a>
        <div
          class="space-x-4"
        >
          <a
            class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
            href="/projects"
          >
            Projects
          </a>
          <a
            class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
            href="/me"
          >
            About Me
          </a>
        </div>
      </div>
    </div>
    <div
      class="container py-10 space-y-6"
    >
      <nav
        class="flex"
      >
        <ol
          class="inline-flex items-center space-x-2"
        >
          <li>
            <a
              class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
              href="/"
            >
              Home
            </a>
          </li>
          <span>
            <svg
              class="w-4 h-4 inline-block leading-4 shrink-0"
              fill="currentColor"
              height="1em"
              role="icon"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.776 1.553a.5.5 0 0 1 .671.223l3 6a.5.5 0 0 1 0 .448l-3 6a.5.5 0 1 1-.894-.448L9.44 8 6.553 2.224a.5.5 0 0 1 .223-.671z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <li>
            <a
              class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
            >
              Blog
            </a>
          </li>
        </ol>
      </nav>
      <div
        class="space-y-3"
      >
        <h1
          class="font-bold font-heading text-4xl"
        >
          Posts (
          1
          )
        </h1>
        <p>
          Tutorials, tricks, personal articles, technical articles, fragments, reference resources and all resources related to the software development.
        </p>
      </div>
      <ul
        aria-label="List of posts"
        class="space-y-1 list-disc list-inside"
        role="list"
      >
        <a
          class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
          href="/blog/how-to-mock-a-rest-api"
        >
          <li>
            How to mock a rest api
          </li>
        </a>
      </ul>
    </div>
    <footer
      class="my-6"
    >
      <div
        class="container"
      >
        <footer
          class="p-4 bg-white rounded-lg shadow flex flex-col md:flex-row md:items-center md:justify-between md:p-6"
        >
          <span
            class="text-gray-500"
          >
            © 2022
             
            <a
              class="hover:underline"
              href="https://yonycalsin.com"
            >
              Yony Calsin™
            </a>
            . All Rights Reserved.
          </span>
          <ul
            class="flex flex-wrap gap-3 items-center justify-center"
          >
            <li>
              <a
                class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
                href="/me"
              >
                About
              </a>
            </li>
            <li>
              <a
                class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
                href="/contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </footer>
      </div>
    </footer>
    <div
      class="space-y-5 fixed bottom-8 left-8 hidden lg:flex flex-col"
    >
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="https://github.com/yonycalsin"
        rel="noreferrer"
        target="_blank"
      >
        Github
      </a>
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="https://www.linkedin.com/in/yonycalsin"
        rel="noreferrer"
        target="_blank"
      >
        Linkedin
      </a>
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="https://twitter.com/yonycalsin"
        rel="noreferrer"
        target="_blank"
      >
        Twitter
      </a>
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="mailto:helloyonycalsin@gmail.com"
        rel="noreferrer"
        target="_blank"
      >
        Email
      </a>
    </div>
  </div>
</body>
`;
